data_collector/collector.py

```diff
@@ -1,49 +1,82 @@
+import logging
+import time
 from opensearchpy import OpenSearch
 from opensearch_dsl import Search, Q
 from datetime import datetime
 
+logger = logging.getLogger(__name__)
 
 class Collector:
     def __init__(self, es_server: str, es_index: str, config: dict):
+        """Init method for instance variables"""
         self.config = config
         self.es_index = es_index
-        self.os_client = OpenSearch(es_server, verify_certs=True, http_compress=True, timeout=30)
+        self.os_client = OpenSearch(es_server, verify_certs=False, http_compress=True, timeout=30)
 
     def collect(self, from_date: datetime, to: datetime):
+        """Collects data from the elastic search"""
+        start_time = time.time()
         data = []
         from_timestamp = from_date.strftime("%Y-%m-%dT%H:%M:%SZ")
         to_timestamp = to.strftime("%Y-%m-%dT%H:%M:%SZ")
+        logger.info(f"Elasticsearch index: {self.es_index}, benchmark: {self.config['benchmark']}")
         query = Q(
             "bool",
-            must_not=[Q("term", **{"jobName.keyword": "garbage-collection"})],
-            must=[Q("range", **{"timestamp": {"gte": from_timestamp, "lte": to_timestamp}})],
+            must_not=[Q("term", **{"jobConfig.name.keyword": "garbage-collection"})],
+            must=[
+                    Q("term", **{"jobConfig.name.keyword": self.config["benchmark"]}),
+                    Q("range", **{"timestamp": {"gte": from_timestamp, "lte": to_timestamp}}),
+                ],
         )
+        logger.debug(f"Constructed Elasticsearch query: {query.to_dict()}")
         s = (
             Search(using=self.os_client, index=self.es_index)
             .filter("term", **{"metricName.keyword": "jobSummary"})
             .query(query)
         )
+
         # Use scan to get all results
         for hit in s.scan():
-            metadata = {}
-            uuid = hit.to_dict().get("uuid")
-            if uuid not in metadata:
-                metadata[uuid] = {"metadata": {}, "metrics": {}}
-            for metadata_field, metadata_value in hit.to_dict().items():
-                if metadata_field in self.config["metadata"]:
-                    metadata[uuid]["metadata"][metadata_field] = metadata_value
-            metrics = self._metrics_by_uuid(uuid)
-            metadata[uuid]["metrics"] = metrics
-            data.append(metadata)
+            run_data = {}
+            jobSummary = hit.to_dict()
+            uuid = jobSummary.get("uuid")
+            if not uuid:
+                logger.warning("Missing UUID in jobSummary, skipping entry.")
+                continue
+            logging.debug(f"Processing UUID: {uuid}")
+            if uuid not in run_data:
+                logger.warning("Missing UUID in run data, skipping entry.")
```

Comment:
This log message is incorrect, Should be something like:

```suggestion
                logger.debug("UUID not present in run data, adding it")
```

Reply:
Updated as suggested.